frontend/src/metabase/admin/settings/auth/components/AuthCard/AuthCard.unit.spec.tsx

```diff
@@ -54,13 +55,31 @@ describe("AuthCard", () => {
 
     expect(props.onDeactivate).toHaveBeenCalled();
   });
-});
 
-const getSetting = (opts?: Partial<AuthSetting>): AuthSetting => ({
-  value: false,
-  ...opts,
+  it("should handle settings set with env vars", () => {
+    const props = getProps({
+      setting: getSetting({
+        value: null,
+        env_name: "MB_JWT_ENABLED",
+        is_env_setting: true,
+      }),
+      isConfigured: true,
+    });
+
+    render(<AuthCard {...props} />);
+
+    expect(screen.getByRole("link")).toHaveTextContent("$MB_JWT_ENABLED");
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
+    expect(screen.queryByTestId("card-badge")).not.toBeInTheDocument();
+  });
 });
 
+const getSetting = (opts?: Partial<AuthSetting>): AuthSetting =>
+  createMockSettingDefinition({
+    value: false,
+    ...opts,
+  }) as AuthSetting;
```

Comment:
I know this is a sin, but I couldn't figure out how to tell TS that this essentially resolves to the same thing without typecasting (AuthSetting is just a Setting Definition with a boolean value, and if one is being passed into getSetting it needs to be boolean anyways).

Reply:
I think obscene type casting in tests is perfectly fine